drivers/npmx/npmx_shell.c

```diff
@@ -3891,17 +3956,22 @@ SHELL_STATIC_SUBCMD_SET_CREATE(
 	SHELL_CMD(debounce, &sub_gpio_debounce, "Debounce configuration", NULL),
 	SHELL_SUBCMD_SET_END);
 
-/* Creating dictionary subcommands (level 4 command) array for command "adc ntc set". */
-SHELL_SUBCMD_DICT_SET_CREATE(adc_ntc_type, cmd_adc_ntc_set,
-			     (ntc_hi_z, NPMX_ADC_NTC_TYPE_HI_Z, "HIGH Z"),
-			     (ntc_10k, NPMX_ADC_NTC_TYPE_10_K, "10k"),
-			     (ntc_47k, NPMX_ADC_NTC_TYPE_47_K, "47k"),
-			     (ntc_100k, NPMX_ADC_NTC_TYPE_100_K, "100k"));
+/* Creating subcommands (level 4 command) array for command "adc ntc type". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_adc_ntc_type,
+			       SHELL_CMD(get, NULL, "Get ADC NTC type", cmd_adc_ntc_type_get),
+			       SHELL_CMD(set, NULL, "Set ADC NTC type", cmd_adc_ntc_type_set),
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 4 command) array for command "adc ntc beta". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_adc_ntc_beta,
+			       SHELL_CMD(get, NULL, "Get ADC NTC beta value", cmd_adc_ntc_beta_get),
+			       SHELL_CMD(set, NULL, "Set ADC NTC beta value", cmd_adc_ntc_beta_set),
+			       SHELL_SUBCMD_SET_END);
 
 /* Creating subcommands (level 3 command) array for command "adc ntc". */
 SHELL_STATIC_SUBCMD_SET_CREATE(sub_adc_ntc,
-			       SHELL_CMD(get, NULL, "Get ADC NTC value", cmd_adc_ntc_get),
-			       SHELL_CMD(set, &adc_ntc_type, "Set ADC NTC value", NULL),
+			       SHELL_CMD(type, &sub_adc_ntc_type, "ADC NTC type", NULL),
```

Comment:
Or perhaps `type` should be renamed to `resistance` everywhere?

Reply:
NTC resistance can be misleading. It can point to actual resistance - based on temperature. But is is a nominal resistance at 25*C. `Type` is more precise.